configure.ac

```diff
@@ -359,7 +359,7 @@ AC_ARG_WITH([variorum],
                 [build Variorum sampler @<:@default=check@:>@])],
                 [AS_IF([test "x$withval" != xyes -a "x$withval" != xno],
                         [VARIORUM_CFLAGS="-I$withval/include"
-                         VARIORUM_LDFLAGS="-L$withval/lib"]
+                         VARIORUM_LDFLAGS="-Wl,-rpath -Wl,$withval/lib -L$withval/lib"]
```

Comment:
We really shouldn't have the -W options here. automake/libtool should be using -rpath of its own under the covers when necessary.

Reply:
I thought so too, but this is the only thing I've found so far that gets the library to link properly. Otherwise "ldd libvariorum_sampler.so" says that libvariorum was not found. Perhaps I am doing something wrong when building, but I believe I am following the steps from the wiki. Do you have any idea what might be going wrong?

Reply:
Well, my first guess is that we should be using _LIBADD rather than LDFLAGS. I'll comment more in the Makefile.am.

Reply:
OH! I am so sorry, it is all coming back to me now and I realize that I should have sent you down a different path. What we want to use (instead of our own manual AC_ARG_WITH) is the very helpful macro AC_LIB_HAVE_LINKFLAGS. There is some documentation on it here:

https://www.gnu.org/software/gnulib/manual/html_node/Searching-for-Libraries.html

But I'll readily admit it can be a little tough to grok that manual.

You'll see that we already use AC_LIB_HAVE_LINKFLAGS in a couple of places. I think that the AC_ARG_ENABLE([papi]) would actually be a perfect example for you. Follow that through to ldms/src/samplers/Makefile.am to see how it uses the results of two AM_CONDITIONALS to decide whether to build a subdirectory. Basically you will wind up doing:

if HAVE_LIBJASSON
if HAVE_LIBVARIORUM
SUBDIR += libvariorum_sampler
endif
endif

And look at ldms/src/samplers/syspapi/Makefile.am. Notice in particular that on the libsyspai_sampler_la_LIBADD line there are "$(LTLIBPAPI) $(LTLIBPFM)".

When you use the AC_LIB_HAVE_LINKFLAGS macro it will automatically create for you the LTLIBVARIORUM and LTLIBJANSSON variables that you can use on the _LIBADD line.

AC_LIB_HAVE_LINKFLAGS([variorum] also auto-magically adds a configure option of the name "--with-libvariorum". It sets a number of other things that we need at well.

I think you would be fine just copying the the AC_ARG_ENABLE([papi] example in the configure.ac file. The only thing that would not do is break out libjansson as an independant test from variorum. But frankly I think it is fine to keep them together at this stage. The ldms team can work out how to split apart the libjansson test at some point in the future when something else wants to use libjansson.